pkg/apis/pipeline/v1alpha1/taskrun_defaults.go

```diff
@@ -29,11 +29,22 @@ func (tr *TaskRun) SetDefaults(ctx context.Context) {
 }
 
 func (trs *TaskRunSpec) SetDefaults(ctx context.Context) {
+	cfg := config.FromContextOrDefaults(ctx)
 	if trs.TaskRef != nil && trs.TaskRef.Kind == "" {
 		trs.TaskRef.Kind = NamespacedTaskKind
 	}
-	cfg := config.FromContextOrDefaults(ctx)
+
+	var timeout *metav1.Duration
+	if IsUpgradeViaDefaulting(ctx) {
+		// This case is for preexisting `TaskRun` before 0.5.0, so let's
```

Comment:
I think I'm missing something: the lemonade key is added to the context at the beginning of the `reconcile` function, so it will be added to old and new `TaskRuns` alike.  However new `TaskRuns` willl have the timeout set, so they won't hit L47.